app/controllers/api/v1/base_controller.rb

```diff
@@ -8,21 +8,35 @@ class BaseController < ApplicationController
       private
 
       def require_authentication!
-        user = User.find(request.headers["X-User-Id"])
+        user = find_user
 
         if user.present? && valid_token?(user.authentication_token)
           sign_in user
         else
-          render status: :unauthorized
+          render json: {}, status: :unauthorized
         end
       end
 
+      def find_user
+        User.find_by(id: user_id_from_headers)
+      end
+
       def valid_token?(token)
+        return unless token && token_from_headers
```

Comment:
This local / global argument mismatch bugs me.

Reply:
what exactly do you mean? 🤔